Adds option to prevent expanding on addMessage()

connormlewis opened the pull request:
When you uncheck the Expand on request box, the view in atom stays collapsed as you navigate the pages that the package is serving

connormlewis committed:
Adds option to prevent expanding on addMessage()

connormlewis committed:
Starts view in collapsed state if !expandOnRequest

leijou commented:
Thanks! Will be good to have this option.

To leave space for future options could you change the expandOnRequest setting to be a dropdown instead of a boolean? Something like:

``` coffee
    expandOnRequest:
      title: 'Expand on request'
      description: 'Expand the server console window when new requests are received by the server'
      type: 'string'
      enum: ['none', 'all']
      default: 'all'
```

I'm imagining in the future this could be extended to only expand for given error levels. Just all or none is good for now though.

connormlewis committed:
Changes expand setting from boolean to string enum

Use in future to force the window to open depending on type of message that is added to the view

leijou commented:
Perfect, thanks a bunch :+1: 

I'll make one small change before releasing, just so that the setting can be modified while the server is running.
